contracts/sumtree-orderbook/src/query.rs

```diff
@@ -214,3 +216,68 @@ pub(crate) fn denoms(deps: Deps) -> ContractResult<DenomsResponse> {
         base_denom: orderbook.base_denom,
     })
 }
+
+pub(crate) fn ticks_by_id(deps: Deps, tick_ids: Vec<i64>) -> ContractResult<TicksResponse> {
+    let ticks = tick_ids
+        .iter()
+        .map(|tick_id| {
+            let tick_state = TICK_STATE
+                .load(deps.storage, *tick_id)
+                .map_err(|_| ContractError::InvalidTickId { tick_id: *tick_id })
+                .unwrap();
+
+            TickIdAndState {
+                tick_id: *tick_id,
+                tick_state,
+            }
+        })
+        .collect();
+
+    Ok(TicksResponse { ticks })
+}
+
+/// Determines the amount of liquidity in unrealized cancels
+fn get_unrealized_cancels(
+    deps: Deps,
+    tick_state: TickState,
+    tick_id: i64,
+) -> ContractResult<TickUnrealizedCancelsState> {
+    let bid_unrealized_cancels = get_root_node(deps.storage, tick_id, OrderDirection::Bid).map_or(
+        tick_state.bid_values.cumulative_realized_cancels,
+        |ask_root| ask_root.get_value(),
+    );
+
+    let ask_unrealized_cancels = get_root_node(deps.storage, tick_id, OrderDirection::Ask).map_or(
+        tick_state.ask_values.cumulative_realized_cancels,
+        |ask_root| ask_root.get_value(),
+    );
+
+    Ok(TickUnrealizedCancelsState {
+        ask_unrealized_cancels,
+        bid_unrealized_cancels,
+    })
+}
+
+// Gets all ticks for the provided vector of IDs and retrieves the value of their sumtree root
+pub(crate) fn ticks_unrealized_cancels_by_id(
+    deps: Deps,
+    tick_ids: Vec<i64>,
+) -> ContractResult<TickUnrealizedCancelsByIdResponse> {
+    let ticks = tick_ids
+        .iter()
+        .map(|tick_id| {
+            let tick_state = TICK_STATE
+                .load(deps.storage, *tick_id)
+                .map_err(|_| ContractError::InvalidTickId { tick_id: *tick_id })
+                .unwrap();
```

Comment:
```suggestion
            let Some(tick_state) = TICK_STATE.may_load(deps.storage, *tick_id)? else {
                return ContractError::InvalidTickId { tick_id: *tick_id };
            };
```

to avoid unwrap and not gobbled up another querier errors